_source/_posts/2021-09-18-apache-apisix-gateway.md

```diff
@@ -0,0 +1,266 @@
+
+First, the user initiates a request, then the gateway itself takes charge of the user authentication process, interacting with the identity provider and sending them an authorization request. The identity provider returns user identity information (user info). After the gateway identifies the user, it forwards the user identity information (user info) to the services in a request header.
+
+{% img blog/apache-apisix-gateway/2021-08-16-2.png alt:"Centralized Identity Authentication work flow" width:"600" %}{: .center-image }
+
+Compared with the traditional authentication mode, centralized identity mode has the following advantages:
+
+1. Centralized auth simplifies the application development process, and reduces the development application workload and maintenance costs by avoiding the need to repeat the development of authentication code for each application.
+2. Centralized authentication mode improves business security. At the gateway level, it can intercept unauthenticated requests in time to protect back-end applications.
+
+## What is OpenID Connect
+
+OpenID Connect (OIDC) is a centralized identity authentication mode. The benefit of using OpenID Connect is that users only need to register and log in with one OpenID Connect identity provider's website and use one account's password information to access different applications. [Okta](https://developer.okta.com/) is a common OpenID Connect identity provider, and the Apache APISIX OpenID Connect plugin supports OpenID.  As a result, the plugin can replace traditional authentication mode with centralized identity authentication.  In this case, we're using Okta.
+
+### OpenID Authentication Process
+
+{% img blog/apache-apisix-gateway/2021-08-16-3.png alt:"OpenID Authentication Process" width:"600" %}{: .center-image }
+
+1. APISIX initiates an authentication request to the Identity Provider.
+2. The user logs in and authenticates on the Identity Provider.
+3. The Identity Provider returns to APISIX with the Authorization Code.
+4. APISIX requests the Identity Provider with the Code extracted from the request parameters.
+5. The Identity Provider sends an answer message to APISIX with the ID Token and Access Token.
+6. APISIX sends the Access Token to the Identity Provider's User Endpoint for authentication.
+7. After passing the authentication, the User Endpoint sends the User Info to APISIX to complete  authentication.
+
+## How to Configure Okta Authentication Using the Apache APISIX OpenID Connect Plug-in
+
+Configuring Okta authentication using the Apache APISIX OpenID Connect plugin is a simple three-step process that allows you to switch from traditional authentication to centralized identity authentication mode. The following sections describe the steps to configure Okta authentication using the OpenID Connect plug-in for Apache APISIX.
```

Comment:
```suggestion
Configuring Okta authentication using the Apache APISIX OpenID Connect plugin is a three-step process that allows you to switch from traditional authentication to centralized identity authentication mode. The following sections describe the steps to configure Okta authentication using the OpenID Connect plugin for Apache APISIX.
```
removed "simple" because it's kind of judgmental, plus it's not really that simple....